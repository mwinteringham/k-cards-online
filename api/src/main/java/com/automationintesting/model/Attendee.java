package com.automationintesting.model;

import com.automationintesting.service.CodeGenerator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class Attendee {

    @JsonProperty
    private String name;
    @JsonProperty
    private String code;

    public Attendee() {}

    public Attendee(String name, String code) {
        this.name = name;
        this.code = code;
    }

    public Attendee(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    @Override
    public String toString() {
        return "Attendee{" +
                "name='" + name + '\'' +
                ", code='" + code +
                '}';
    }
}
