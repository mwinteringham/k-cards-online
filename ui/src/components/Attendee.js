import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Button from 'react-bootstrap/Button';
import Alert from 'react-bootstrap/Alert';
import API from '../api/api';
import { useState } from 'react';
import { useGlobalState } from './state';

function Attendee(){

    const [showConfirm, setConfirm] = useState(false);
    const [workshopCode] = useGlobalState('workshopCode');

    async function sendCard(cardType){
        console.log(workshopCode);
        const res = await API.sendCard(cardType);

        if(res.statusCode === 201){
            setConfirm(true);
        }
    } 

    return(
        <div>
            {(showConfirm && <Alert variant="success" onClose={() => setConfirm(false)} dismissible><Alert.Heading>Your card has been received by the Host</Alert.Heading></Alert>)}
            <Row>
                <Col>
                    <Button data-testid='greenCard' className='btn-success' onClick={() => sendCard('green')} style={{ width: '100%', height: '8rem'}}>Send Green Card</Button>
                </Col>
                <Col>
                    <Button data-testid='yellowCard' className='btn-warning' onClick={() => sendCard('yellow')} style={{ width: '100%', height: '8rem' }}>Send Yellow Card</Button>
                </Col>
                <Col>
                    <Button data-testid='redCard' className='btn-danger' onClick={() => sendCard('red')} style={{ width: '100%', height: '8rem' }}>Send Red Card</Button>
                </Col>
            </Row>
            <Row className='mt-4'>
                <Col>
                    <Button className='btn-secondary' style={{ width: '100%', height: '4rem'}}>Leave Workshop</Button>
                </Col>
            </Row>
        </div>
    );
}

export default Attendee