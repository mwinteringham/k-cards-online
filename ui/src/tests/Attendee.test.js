import { render, fireEvent, screen, waitFor } from '@testing-library/react';
import Attendee from '../components/Attendee';

test('renders the Attendee component', () => {
    const attendeeComponent = render(<Attendee />);
    
    expect(attendeeComponent.baseElement).toMatchSnapshot();
});

test('confirm message appears when sending a card', async () => {
    const attendeeComponent = render(<Attendee />);
    
    fireEvent.click(screen.getByTestId(/greenCard/i));
    await waitFor(() => screen.getByText('Your card has been received by the Host')); 

    expect(attendeeComponent.baseElement).toMatchSnapshot();
});

