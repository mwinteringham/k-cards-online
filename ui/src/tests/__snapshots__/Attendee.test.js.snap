// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`confirm message appears when sending a card 1`] = `
<body>
  <div>
    <div>
      <div
        class="fade alert alert-success alert-dismissible show"
        role="alert"
      >
        <button
          class="close"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
          <span
            class="sr-only"
          >
            Close alert
          </span>
        </button>
        <div
          class="alert-heading h4"
        >
          Your card has been received by the Host
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <button
            class="btn-success btn btn-primary"
            data-testid="greenCard"
            style="width: 100%; height: 8rem;"
            type="button"
          >
            Send Green Card
          </button>
        </div>
        <div
          class="col"
        >
          <button
            class="btn-warning btn btn-primary"
            data-testid="yellowCard"
            style="width: 100%; height: 8rem;"
            type="button"
          >
            Send Yellow Card
          </button>
        </div>
        <div
          class="col"
        >
          <button
            class="btn-danger btn btn-primary"
            data-testid="redCard"
            style="width: 100%; height: 8rem;"
            type="button"
          >
            Send Red Card
          </button>
        </div>
      </div>
      <div
        class="mt-4 row"
      >
        <div
          class="col"
        >
          <button
            class="btn-secondary btn btn-primary"
            style="width: 100%; height: 4rem;"
            type="button"
          >
            Leave Workshop
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`renders the Attendee component 1`] = `
<body>
  <div>
    <div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <button
            class="btn-success btn btn-primary"
            data-testid="greenCard"
            style="width: 100%; height: 8rem;"
            type="button"
          >
            Send Green Card
          </button>
        </div>
        <div
          class="col"
        >
          <button
            class="btn-warning btn btn-primary"
            data-testid="yellowCard"
            style="width: 100%; height: 8rem;"
            type="button"
          >
            Send Yellow Card
          </button>
        </div>
        <div
          class="col"
        >
          <button
            class="btn-danger btn btn-primary"
            data-testid="redCard"
            style="width: 100%; height: 8rem;"
            type="button"
          >
            Send Red Card
          </button>
        </div>
      </div>
      <div
        class="mt-4 row"
      >
        <div
          class="col"
        >
          <button
            class="btn-secondary btn btn-primary"
            style="width: 100%; height: 4rem;"
            type="button"
          >
            Leave Workshop
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
